pkg/provider-local/machine-provider/local/create_machine.go

```diff
@@ -65,6 +65,10 @@ func (d *localDriver) CreateMachine(ctx context.Context, req *driver.CreateMachi
 	return &driver.CreateMachineResponse{
 		ProviderID: pod.Name,
 		NodeName:   pod.Name,
+		Addresses: []corev1.NodeAddress{{
+			Type:    corev1.NodeInternalIP,
+			Address: pod.Status.PodIP,
```

Comment:
If the pod has multiple IPs (dual-stack) we might want to return multiple addresses based on `pod.Status.PodIPs`.